tests/main/upgrade-from-release/task.yaml

```diff
@@ -62,7 +62,13 @@ execute: |
     snap install go-example-webserver
     tests.systemd wait-for-service -n 30 --state active snap.go-example-webserver.webserver.service
     snap install test-snapd-tools
-    test-snapd-tools.echo hello | MATCH hello
+    # In case of known error on older snapd deb versions e.g. 2.32.5+18.04,
+    # rerun with SNAP_REEXEC=1
+    if result=$(test-snapd-tools hello | MATCH hello); then
```

Comment:
Shouldn't this condition be negated? It should only take this branch if it did not match hello. Also shouldn't it be `test-snapd-tools.echo`? Finally, it looks like the "cannot perform readlinkat()..." is over stderr (as well as "re-exec disabled by user") and `MATCH` uses grep with the `-E` flag so you would need to escape the parenthesis. So I believe the portion of code should be: 

```
if result=$(test-snapd-tools.echo hello 2>&1); then
  MATCH hello <<< "$result"
else
  SNAPD_DEBUG=1 snap version 2>&1 | MATCH "re-exec disabled by user"
  MATCH "cannot perform readlinkat\(\) on the mount namespace file descriptor of the init process" <<< "$result"
  SNAP_REEXEC=1 test-snapd-tools.echo hello | MATCH hello
fi
```

Reply:
Apologies, this somehow went untested! Your suggestion is perfect, thanks.